pandas/core/internals/blocks.py

```diff
@@ -2349,11 +2349,21 @@ def external_values(values: ArrayLike) -> ArrayLike:
     proper extension array).
     """
     if isinstance(values, (PeriodArray, IntervalArray)):
+        warnings.warn(
+            "series.values will stop converting tz from dt64tz, interval to object and period to object",
+            FutureWarning,
+            stacklevel=find_stack_level(),
+        )
         return values.astype(object)
     elif isinstance(values, (DatetimeArray, TimedeltaArray)):
         # NB: for datetime64tz this is different from np.asarray(values), since
         #  that returns an object-dtype ndarray of Timestamps.
         # Avoid raising in .astype in casting from dt64tz to dt64
+        warnings.warn(
+            "series.values will stop converting tz from dt64tz, interval to object and period to object",
+            FutureWarning,
+            stacklevel=find_stack_level(),
+        )
```

Comment:
If the data is tz-agnostic, then there will be no behavior change. So users should not see a warning.

Reply:
How can I see if the data is tz-agnostic?

Reply:
I believe `isinstance(values.dtype, pd.DatetimeTZDtype)`